### ISSUE TITLE
Model Interaction for Unity

### ISSUE THREAD

**Issue by Hai-Hoang-88** [csl-hcmc/SaiGon-Peninsula#15]
We need to coordinate how we can setup interaction model.

Here is the link for Unity model when we import part by part from Cityengine
The unity model is done by @viettham1998 
[Unity model](urhttps://github.com/csl-hcmc/SaiGon-Peninsula/tree/main/Software/2.Unityl) 

@Carsonsmuts please keep this discussion going here to keep tracking

**Comment by Carsonsmuts** [comment]
Hi @Hai-Hoang-88 , sorry I missed this :)

Are you referring to the interaction between ViKi and Unity3D, or CityScope and Unity3D. 

C

**Comment by Hai-Hoang-88** [comment]
The interaction I mentioned was about changing some parameters (currently number of floors for a realistic model) and a kick starter to host the model (non-interact and interact).

We are working on the landscape, and material for the realistic model (try to finish what is almost ready). 
And need to make a model which matches the grid? later with the more time-consuming task.

**Comment by Carsonsmuts** [comment]
Do you currently have the floors modelled? (ie. "currently number of floors for a realistic model")
Are the floors ready to be used?

**Comment by Hai-Hoang-88** [comment]
Yes, it is. Those floor models are in the [Resource folder](https://github.com/csl-hcmc/SaiGon-Peninsula/tree/main/Software/2.Unity/Assets/Resources), 

Those models have 2 "types": floor models and envelope models.  
Script has worked with floor models but envelope one still has issued when scaling in the vertical direction.

**Comment by Carsonsmuts** [comment]
So currently the issue is with the envelope, correct? 

How is the script modifying the envelope exactly :

1. Stretching a mesh model?
2. Swapping out static library parts?
3. Generating an envelope parametricly on demand?

**Comment by Hai-Hoang-88** [comment]
We go with a simple step firstly by stretching a mesh model.
As I imagine, the interaction will go as follows:

1. Showing static model (A) with details
2. Interact with simpler version (A-) with polygon massing  (var: number of floors)
3. Interact with Grid (B) with Rectangle Massing (var: number of floors, number of houses, and location of legos).

Please lets me know your thought
